Johnsmartco patch2 (#127)
k8ssandra/k8ssandra: docs/content/en/docs/topics/restore-a-backup/_index.md

* K8ssandra-cluster Helm chart, which we'll extend with `backupRestore` Medusa buckets for Amazon S3 integration
* Sample files in GitHub:
  * [medusa-bucket-key.yaml](./medusa-bucket-key.yaml) to create a secret with credentials for AWS S3 buckets

>   * [backup-restore-values.yaml](./backup-restore-values.yaml) to enable Medusa (backup/restore service) and set related minimal values

Author: Done, thanks

You will need storage for the backups. This topic shows the use of AWS S3 buckets.

* If you'll use AWS S3, before proceeding with the configuration described below, verify that you know the `aws_access_key_id` and `aws_secret_access_key` values. Or  contact your IT team if they manage those assets. You'll provide those details in an edited version of the [medusa-bucket-key.yaml](medusa-bucket-key.yaml) file. For information about the S3 setup steps, see this helpful [readme](https://github.com/thelastpickle/cassandra-medusa/blob/master/docs/aws_s3_setup.md).  

* Add and update the following repo, which has in one chart all the settings for K8ssandra plus the backup/restore settings:

```
helm repo add k8ssandra https://helm.k8ssandra.io/
helm repo update
```

```
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "k8ssandra" chart repository
Update Complete. ⎈Happy Helming!⎈
```

### Create secret for read/write access to an S3 bucket

> Before installing the k8ssandra cluster, we need to supply credentials so that Medusa has read/write to an AWS S3 bucket, which is where the backup will be stored.  Currently, Medusa supports local, Amazon S3, Google Cloud Storage, and Azure buckets. Currently, K8ssandra supports S3. 

Reviewer: ```suggestion
Before installing the k8ssandra cluster, we need to supply credentials so that Medusa has read/write to an AWS S3 bucket, which is where the backup will be stored.  Medusa currently supports local, Amazon S3, Google Cloud Storage, and Azure buckets. K8ssandra only exposes configuration for Amazon S3 at this time.
```

Reviewer 2: There are couple problems I spotted. Here is a revised version:

Before installing the k8ssandra cluster, we need to supply credentials so that Medusa has read/write access to an AWS S3 bucket, which is where the backup will be stored.  Medusa currently supports local, Amazon S3, Google Cloud Storage, and Azure Blob Storage.. K8ssandra exposes configuration for Amazon S3 and Google Cloud Storage at this time.

Reviewer: Updated suggestion with @jsanda's recommendation

```suggestion
Before installing the k8ssandra cluster, we need to supply credentials so that Medusa has read/write access to an AWS S3 bucket, which is where the backup will be stored. Medusa currently supports local, Amazon S3, Google Cloud Storage, Azure Blob Storage, and IBM Cloud Object Storage. K8ssandra exposes configuration for Amazon S3 and Google Cloud Storage at this time.
```

Reviewer: @johnsmartco if this works for you commit the suggestion and we will wrap up approvals. Thanks!

> Before installing the k8ssandra cluster, we need to supply credentials so that Medusa has read/write to an AWS S3 bucket, which is where the backup will be stored.  Currently, Medusa supports local, Amazon S3, Google Cloud Storage, and Azure buckets. Currently, K8ssandra supports S3. 

Reviewer 2: This still says `has read/write to an AWS S3 bucket`. It should be `has read/write access to an AWS S3 bucket`.

> Before installing the k8ssandra cluster, we need to supply credentials so that Medusa has read/write to an AWS S3 bucket, which is where the backup will be stored.  Currently, Medusa supports local, Amazon S3, Google Cloud Storage, and Azure buckets. Currently, K8ssandra supports S3. 

Reviewer: ```suggestion
Before installing the k8ssandra cluster, we need to supply credentials so that Medusa has read/write access to an AWS S3 bucket, which is where the backup will be stored.  Currently, Medusa supports local, Amazon S3, Google Cloud Storage, and Azure buckets. Currently, K8ssandra supports S3.
```

* If you'll use AWS S3, before proceeding with the configuration described below, verify that you know the `aws_access_key_id` and `aws_secret_access_key` values. Or  contact your IT team if they manage those assets. You'll provide those details in an edited version of the [medusa-bucket-key.yaml](./medusa-bucket-key.yaml) file. For information about the S3 setup steps, see this helpful [readme](https://github.com/thelastpickle/cassandra-medusa/blob/master/docs/aws_s3_setup.md).  

* If you haven't already, add and update the following repo, which has in one chart all the settings for K8ssandra plus the backup/restore settings:

```
helm repo add k8ssandra https://helm.k8ssandra.io/
helm repo update
```

```
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "k8ssandra" chart repository
Update Complete. ⎈Happy Helming!⎈
```

> * If you haven’t already, install the k8ssandra chart in any namespace; this command will use the `default` namespace.

Reviewer: `If you haven’t already,`

Could we rephrase this as it is the same as the previous item? I'm on the fence regarding this change request so feel free to push back.

Author: Done, thanks

### Create secret for read/write access to an S3 bucket

> Before creating the k8ssandra-cluster, we need to supply credentials so that Medusa has read/write to an AWS S3 bucket, which is where the backup will be stored.  Currently, Medusa supports local, Amazon S3, GKE, and Azure buckets. Currently, K8ssandra supports S3. 

Reviewer: `GKE` should be `Google Cloud Storage`.

Author: Done, thanks Chris.

secret/medusa-bucket-key configured
```

### Create or update the k8ssandra cluster

Install the `k8ssandra` chart with the following properties. You can reference the provided [backup-restore-values.yaml](./backup-restore-values.yaml) file. It contains:

```
size: 3
backupRestore: 
  medusa:
    enabled: true
    bucketName: k8ssandra-bucket-dev
    bucketSecret: medusa-bucket-key

>     multiTenant: true

Author: Ok will do in the sample file and in this topic's list of its content - thanks

>     multiTenant: true

Author: Done in topic and sample file

    storage: s3
```

> The chart's entries relate to a Kubernetes Secret, which contains the object store credentials. 

Reviewer 2: More specifically the `bucketSecret` property specifies the name of a secret that should contain an AWS access key. The AWS account with which the key is associated should have the permissions as described in the [Medusa docs](https://github.com/thelastpickle/cassandra-medusa/blob/master/docs/aws_s3_setup.md) that are required for Medusa to access the S3 bucket.

Author: Incorporated - thanks John

Example:

`helm install k8ssandra k8ssandra/k8ssandra -f backup-restore-values.yaml`

Allow a few minutes for the pods to start and proceed to a Ready state; check the pod status periodically:

```
kubectl get pods                              
NAME                                                         READY   STATUS    RESTARTS   AGE
cass-operator-86d4dc45cd-8p7cq                               1/1     Running   0          98s
k8ssandra-tools-kube-prome-operator-6bcdf668d4-b2r6v         1/1     Running   0          98s
.
.
.
```

Backup and restore operations are enabled by default. In the example YAML, `bucketName` corresponds to the name of the S3 bucket: `K8ssanda-bucket-dev`.  The `bucketSecret` corresponds to the secret credentials.

The `k8ssandra` Helm chart includes the Grafana Operator. Notice that `k8ssandra` adds a number of properties in the `cassdc` datacenter.  

> `kubectl get cassdc dc1 -o yaml`

Reviewer 2: Checking for `cassandrOperatorProgress: Ready` is not always sufficient. The CassandraDatacenter status reports several conditions. The user should also look for the following condition in the status:

```yaml
status:
  cassandraOperatorProgress: Ready
  conditions:
  ...
  - lastTransitionTime: "2021-02-03T17:04:52Z"
    message: ""
    reason: ""
    status: "True"
    type: Ready
  ...
```

Author: Incorporated - thanks

`kubectl get pods`                           

```
NAME                                                  READY   STATUS             RESTARTS   AGE
k8ssandra-cass-operator-66dc79dd97-kvzdq              1/1     Running            44         18h
k8ssandra-dc1-default-sts-0                           1/2     Running            0          16m
k8ssandra-grafana-7cb58d4f6b-5dn6g                    2/2     Running            1          18h
k8ssandra-kube-prometheus-operator-85695ffb-xdq8c     1/1     Running            0          18h
k8ssandra-medusa-operator-58586cb9c4-qctvb            1/1     Running            5          16m
k8ssandra-reaper-operator-97754757-p7rz8              1/1     Running            11         18h
prometheus-k8ssandra-kube-prometheus-prometheus-0     2/2     Running            1          18h
prometheus-pulsar-kube-prometheus-sta-prometheus-0    2/2     Running            1          18h
```

Backup and restore operations are enabled by default. In the example YAML, `bucketName` corresponds to the name of the S3 bucket: `K8ssanda-bucket-dev`.  The `bucketSecret` corresponds to the secret credentials.

> The `k8ssandra` Helm chart includes the Grafana Operator. Notice that `k8ssandra` adds a number of properties in the `cassdc` datacenter.  

Reviewer 2: We no longer include the Grafana operator. We will include Grafana itself though.

Author: Edited - thanks

### Create or update the k8ssandra-cluster

Install the `k8ssandra-cluster` chart with the following properties. You can reference the provided [backup-restore-values.yaml](./backup-restore-values.yaml) file. It contains:

```
size: 3
backupRestore: 
  medusa:
    enabled: true
    bucketName: k8ssandra-bucket-dev
    bucketSecret: medusa-bucket-key
    multiTenant: true
    storage: s3
```

Example:

`helm install k8ssandra-cluster-1 k8ssandra/k8ssandra-cluster -f backup-restore-values.yaml`

> Backup and restore operations are enabled by default. In the example YAML, `bucketName` corresponds to the name of the S3 bucket: `K8ssanda-bucket-dev`.  The `bucketSecret` corresponds to the secret credentials.

Reviewer: > The `bucketSecret` corresponds to the secret credentials.

We should clarify that it relates to a Kubernetes Secret which contains the object store credentials. Bonus points if we illustrate what that secret should look like.

Author: Done, thanks

> The `k8ssandra-cluster` Helm chart also includes cluster services and the Grafana Operator. Notice that `k8ssandra-cluster` adds a number of properties in the `cassdc` datacenter.  

Reviewer: What do we mean by "cluster services"?

Author: Reworded, thanks

```
NAME               	NAMESPACE	REVISION	UPDATED                             	STATUS  	CHART                  	APP VERSION

> k8ssandra.         	default  	1       	2021-01-16 20:17:23.107265 -0700 MST	deployed	k8ssandra-0.9.0        	3.11.7  

Reviewer 2: Should the period after _k8ssandra_ in the first column be there?

Author: good catch - removed
